### PR TITLE
Remove session middleware and simplify session handling code

### PR DESCRIPTION
**Pull request by jerben**
This is an approach without using middlewares. It simplifies quite a bit, what do you think?
We could use the same approach for the `htmx` and maybe the `form` middlewares,

### CODE REVIEW

**Comment by aronerben** [comment]
> This is an approach without using middlewares. It simplifies quite a bit, what do you think?
> We could use the same approach for the `htmx` and maybe the `form` middlewares,

I'm a big fan, great refactor!

**Comment by jerben** [comment]
Conflicts resolved

**Review by aronerben — APPROVED** [review]
Looks good!

**Review by aronerben — CHANGES_REQUESTED** [review]
Please check

File: sihl/src/sihl.mli [hunk 1]
```diff
@@ -200,26 +209,6 @@ module Web : sig
     val htmx : Rock.Middleware.t
     val id : Rock.Middleware.t
     val json : Rock.Middleware.t
-
-    (** [session ?cookie_key ?secret ()] returns a middleware that reads and
```
[the remark this answers is not in the document]
Comment (jerben) [reply sihl/src/sihl.mli:188]: I agree: https://oxidizing.github.io/sihl/sihl/index.html#limitations

File: sihl/src/sihl.mli [hunk 2]
```diff
@@ -109,10 +109,19 @@ module Web : sig
...
+      -> string
+      -> Opium.Request.t
+      -> string option
 
-    val find : string -> Opium.Request.t -> string option
-    val set : string * string option -> Opium.Response.t -> Opium.Response.t
+    val set
+      :  ?cookie_key:string
+      -> ?secret:string
+      -> (string * string) list
```
[the remark this answers is not in the document]
Comment (jerben) [reply sihl/src/sihl.mli:107]: Fixed

File: sihl/src/sihl.mli [hunk 3]
```diff
@@ -109,10 +109,19 @@ module Web : sig
   end
 
   module Session : sig
-    exception Session_not_found
+    val find
```
[the remark this answers is not in the document]
Comment (jerben) [reply sihl/src/sihl.mli:97]: Fixed

**Comment by jerben** [comment]
I also updated the documentation

File: sihl/src/sihl.mli (hunk 1 above)
[the remark this answers is not in the document]
Comment (aronerben) [reply sihl/src/sihl.mli:188]: Ah :)

File: sihl/src/sihl.mli (hunk 3 above)
[the remark this answers is not in the document]
Comment (aronerben) [reply sihl/src/sihl.mli:105]: Thanks

File: sihl/src/sihl.mli (hunk 2 above)
[the remark this answers is not in the document]
Comment (aronerben) [reply sihl/src/sihl.mli:125]: Thanks

**Review by aronerben — APPROVED** [review]

**Pull request merged**